File: 프로그래머스/0/181880. 1로 만들기/1로 만들기.java
class Solution {
    public int solution(int[] num_list) {
        int answer = 0;
        
        for(int num : num_list) {
            while(num != 1){
                num = num % 2 == 0 ? num / 2 : (num - 1) / 2; // 참고: int 타입이라 / 연산 시 -1을 안해도 절삭됨
                answer++;
            }
        }
        
        return answer;
    }
}

// 다른 풀이
return Arrays.stream(num_list)
             .map(i -> Integer.toBinaryString(i).length() - 1)
             .sum();

File: 프로그래머스/0/181880. 1로 만들기/1로 만들기.py
def solution(num_list):
    answer = 0
    
    for num in num_list:
        while num != 1:
            if num % 2 == 0:
                num = num / 2
                answer += 1
            else:
                num = (num - 1) / 2
                answer += 1
    
    return answer

# 다른 풀이
return sum(len(bin(i)) - 3 for i in num_list)

# 다른 풀이
 answer = 0

for n in num_list:
    while n != 1:
        n //= 2
        answer += 1

return answer

